core/src/main/java/com/linecorp/armeria/common/logging/JsonLogFormat.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.util.TextFormatter;
+
+/**
+ * A formatter that convert {@link RequestLog} into json format message.
+ */
+public class JsonLogFormat implements LogFormat {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static String toJson(LinkedHashMap<String, String> contents) {
+        try {
+            return objectMapper.writeValueAsString(contents);
```

Comment:
Could we use `JsonGenerator` instead of keeping the entire object in a map? e.g.

```
StringWriter writer = new StringWriter(someReasonableCapacity);
JsonGenerator gen = objectMapper.createGenerator(writer);
gen.writeStartObject();
...
gen.writeEndObject();
return writer.toString();
```